pkg/controllers/vdb/createarchive_reconciler.go

```diff
@@ -0,0 +1,203 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vdb
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/createarchive"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/saverestorepoint"
+	"github.com/vertica/vertica-kubernetes/pkg/vdbstatus"
+	"github.com/vertica/vertica-kubernetes/pkg/version"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateArchiveReconciler will create archive and save restore point if triggered
+type CreateArchiveReconciler struct {
+	VRec *VerticaDBReconciler
+	Vdb  *vapi.VerticaDB
+	Log  logr.Logger
+	client.Client
+	Dispatcher     vadmin.Dispatcher
+	PFacts         *PodFacts
+	OriginalPFacts *PodFacts
+	InitiatorIP    string // The IP of the pod that we run vclusterOps from
+}
+
+func MakeCreateArchiveReconciler(r *VerticaDBReconciler, vdb *vapi.VerticaDB, log logr.Logger,
+	pfacts *PodFacts, dispatcher vadmin.Dispatcher, cli client.Client) controllers.ReconcileActor {
+	pfactsForMainCluster := pfacts.Copy(vapi.MainCluster)
+	return &CreateArchiveReconciler{
+		VRec:           r,
+		Log:            log.WithName("CreateArchiveReconciler"),
+		Vdb:            vdb,
+		Client:         cli,
+		Dispatcher:     dispatcher,
+		PFacts:         &pfactsForMainCluster,
+		OriginalPFacts: pfacts,
+	}
+}
+
+// Reconcile will create an archive if the status condition indicates true
+// And will save restore point to the created arcihve if restorePoint.archive value
+// is provided in the CRD spec
+func (c *CreateArchiveReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
```

Comment:
You should also check early if it is a vclusterops deployment and and log an error if it isn't.

Reply:
I added a check in https://github.com/vertica/vertica-kubernetes/pull/918/commits/1233fa9c073cc48451c8551bf987a116d8121677
Not sure if this is the checked you mentioned. 
And since this check method simply return bool value. But the log error method need an error object, I use the fmt.error and give a msg. 
c.Log.Error(fmt.Errorf("invalid deployment method"), "create archive failed, not a vclusterops deployment")

Reply:
Sorry I meant a normal warning `log.info`

Reply:
Yes but a `log.info` is enough